File: chapter-20-execution-interceptor/code-java/src/main/java/com/example/demo/config/JooqConfig.java
package com.example.demo.config;

import org.jooq.impl.DefaultExecuteListenerProvider;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JooqConfig {

    @Bean
    public DefaultExecuteListenerProvider performanceListenerProvider() {
        return new DefaultExecuteListenerProvider(new PerformanceListener());
    }
}

File: chapter-20-execution-interceptor/code-kotlin/src/main/kotlin/com/example/demo/config/JooqConfig.kt
package com.example.demo.config

import org.jooq.impl.DefaultExecuteListenerProvider
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class JooqConfig {

    @Bean
    fun performanceListenerProvider(): DefaultExecuteListenerProvider {
        return DefaultExecuteListenerProvider(PerformanceListener())
    }
}
